net/node/chanqueue.go

```diff
@@ -0,0 +1,10 @@
+package node
+
+type ChanQueue chan struct{}
+
+func MakeChanQueue(n int) ChanQueue {
+	return make(chan struct{}, n)
+}
+
+func (s ChanQueue) acquire() { s <- struct{}{} }
```

Comment:
acquire and release make me think of lock rather than channel. Is there a better name for this? or is this naming common?